core/sql/sqlcat/TrafDDLdesc.h

```diff
@@ -0,0 +1,1120 @@
+  virtual unsigned char getClassVersionID()
+  {
+    return 1;
+  }
+
+  virtual void populateImageVersionIDArray()
+  {
+    setImageVersionID(0,getClassVersionID());
+  }
+
+  virtual short getClassSize()      { return (short)sizeof(TrafHistIntervalDesc); }
+ 
+  virtual Long pack(void *space);
+  virtual Lng32 unpack(void * base, void * reallocator);
+
+  virtual TrafHistIntervalDesc *histIntervalDesc() const { return (TrafHistIntervalDesc*)this; }
+
+  char*   histid;
+  char*   intboundary;
+
+  Float32  rowcount;
+  Float32  uec;
+
+  Int32   intnum;
+  Int32   fillerInt32;
+
+  Int64 histIntervalDescFlags; // my flags
+
+  char filler[16];
+};
+
+class TrafIndexesDesc : public TrafDesc {
+public:
+  TrafIndexesDesc() : TrafDesc(DESC_INDEXES_TYPE)
+  {}
+
+  // ---------------------------------------------------------------------
+  // Redefine virtual functions required for Versioning.
+  //----------------------------------------------------------------------
+  virtual unsigned char getClassVersionID()
+  {
+    return 1;
+  }
+
+  virtual void populateImageVersionIDArray()
+  {
+    setImageVersionID(0,getClassVersionID());
+  }
+
+  virtual short getClassSize()      { return (short)sizeof(TrafIndexesDesc); }
+ 
+  virtual Long pack(void *space);
+  virtual Lng32 unpack(void * base, void * reallocator);
+
+  virtual TrafIndexesDesc *indexesDesc() const { return (TrafIndexesDesc*)this; }
+
+  enum IndexesDescFlags
+    { 
+      SYSTEM_TABLE_CODE = 0x0001,  
+      EXPLICIT          = 0x0002,
+      VOLATILE          = 0x0004,
+      IN_MEM_OBJ        = 0x0008,
+      UNIQUE            = 0x0010
+    };
+
+  void setSystemTableCode(NABoolean v) 
+  {(v ? indexesDescFlags |= SYSTEM_TABLE_CODE : indexesDescFlags &= ~SYSTEM_TABLE_CODE); };
+  NABoolean isSystemTableCode() { return (indexesDescFlags & SYSTEM_TABLE_CODE) != 0; };
+
+  void setExplicit(NABoolean v) 
+  {(v ? indexesDescFlags |= EXPLICIT : indexesDescFlags &= ~EXPLICIT); };
+  NABoolean isExplicit() { return (indexesDescFlags & EXPLICIT) != 0; };
+
+  void setVolatile(NABoolean v) 
+  {(v ? indexesDescFlags |= VOLATILE : indexesDescFlags &= ~VOLATILE); };
+  NABoolean isVolatile() { return (indexesDescFlags & VOLATILE) != 0; };
+
+  void setInMemoryObject(NABoolean v) 
+  {(v ? indexesDescFlags |= IN_MEM_OBJ : indexesDescFlags &= ~IN_MEM_OBJ); };
+  NABoolean isInMemoryObject() { return (indexesDescFlags & IN_MEM_OBJ) != 0; };
+
+  void setUnique(NABoolean v) 
+  {(v ? indexesDescFlags |= UNIQUE : indexesDescFlags &= ~UNIQUE); };
+  NABoolean isUnique() { return (indexesDescFlags & UNIQUE) != 0; };
+
+  ComPartitioningScheme partitioningScheme() 
+  { return (ComPartitioningScheme)partitioningScheme_; }
+  void setPartitioningScheme(ComPartitioningScheme v) 
+  { partitioningScheme_ = (Int16)v; }
+  ComRowFormat rowFormat() { return (ComRowFormat)rowFormat_; }
+  void setRowFormat(ComRowFormat v) { rowFormat_ = (Int16)v; }
+
+  char* tablename;  // name of the base table
```

Comment:
Right. The 'char *' are pointing to char arrays and not any classes.
Currently, all classes either have native fields (int, short, etc), or 'char *' pointers
or 'TrafDesc' pointers.
If any pointer  in future points to a class, then it will need to include
pack/unpack of the class it is pointing to.